### PR TITLE
Allure reporter add environment variables in config

### LINKED ISSUE #10321: [🐛 Bug]: Allure reporter adding environment value overwrites the last value
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

v16.13.2

### Mode

WDIO Testrunner

### Which capabilities are you using?

_No response_

### What happened?

When using allureReporter.addEnvironment() the new value overwrites the last value.

The problem comes from https://github.com/webdriverio/webdriverio/blob/f0c607340b4f31a4ae48509115335283f488e38f/packages/wdio-allure-reporter/src/reporter.ts#L740.

I guess the reason is that in allure reporter you should now set all environment variables at once as it is a `Record<string,string>`, which makes IMHO also sense.

### What is your expected behavior?

I have two approaches to solve the problem, and I'm happy to implement whichever one the community prefers.

1. Setting all variables in wdio.conf in the reporter configuration is the first approach. Normally, you would not/should not change environment variables during the tests. So IMHO something like this makes sense:
```
reporters: [['allure', {
    outputDir: 'allure-results',
    environment: [
        a:'a',
        b:'b'
    ]
}]],
```
2. Changing `addEnvironment(name: string, value: string)` to `setEnvironment(Record<string,string>)` is the second approach.

The second one is more flexible, but in my opinion, the first one is the right approach because the second one opens the door to mistakes and makes it possible to overwrite the environment variable information.

### How to reproduce the bug.

Just write this anywhere in your test:
```
allureReporter.addEnvironment('d', 'c');
allureReporter.addEnvironment('a', 'b');
```
### Relevant log output

```typescript
a b
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by m4hdyar**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
Closes #10321 
With this pull request we can use  `reportedEnvironmentVars` in the config instead of using `addEnvironment` in tests

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [X] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
Thanks for raising the issue.

I am a bit concerned removing `addEnvironment` entirely because it could potentially break other peoples test. I would recommend better deprecate it and log a warning that the method will be removed soon. Does `addEnvironment` still has a function in the reporter?

**Comment by m4hdyar** [comment]
> Thanks for raising the pull request.
> 
> I am a bit concerned removing `addEnvironment` entirely because it could potentially break other peoples test. I would recommend better deprecate it and log a warning that the method will be removed soon. Does `addEnvironment` still has a function in the reporter?

`addEnvironment` still has a function, but when you use it a second time, it will overwrite the first value. This means you cannot add more than one environment variable. So, if you only need to add one environment variable, it works correctly, although that is not usually the case. 
If we keep it, it will also override environment variables that you write in the config.

I understand the importance of not disrupting people's tests. But the error is relatively easy to fix. However, in my opinion, deprecating addEnvironment without correcting its functionality would result in incorrect functionality without displaying any error, leading to unpredictable behavior for users.

On the other hand, if we choose to deprecate and also correct the behavior of `addEnvironment`, it would involve introducing a significant amount of unnecessary complexity. 

What do you think?

**Comment by christian-bromann** [comment]
> However, in my opinion, deprecating addEnvironment without correcting its functionality would result in incorrect functionality without displaying any error, leading to unpredictable behavior for users.

Why we can't deprecated and notify people at the same time about the error, even give them some instructions how to use the new interface. I am not sure why keeping `addEnvironment` imposes more complexity but I would rather want to keep it with a deprecation warning.

**Comment by BorisOsipov** [comment]
@m4hdyar @christian-bromann

The best solution for me would be to deprecate addEnvironment and display a warning message with a link to the documentation. Additionally, we should modify addEnvironment to have no functionality at all. This approach will alert users to switch to the new method and provide an easy way to migrate. Of course, we may encounter some issues where users complain that addEnvironment is not working, but there is no better solution available.

**Comment by christian-bromann** [comment]
Sounds reasonable @BorisOsipov  👍

File: packages/wdio-allure-reporter/src/common/api.ts [hunk 1]
```diff
@@ -144,11 +144,11 @@ export function addTag(tag: string) {
...
  * Add environment value
  * @name addEnvironment
- * @param {string} name - environment name
- * @param {string} value - environment value
+ * @deprecated addEnvironment is deprecated. Use reportedEnvironmentVars in config instead.
  */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 export function addEnvironment (name: string, value: string) {
-    tellReporter(events.addEnvironment, { name, value })
+    console.warn('addEnvironment is deprecated and has no longer any functionality. Use reportedEnvironmentVars in wdio config instead. Read more in https://webdriver.io/docs/allure-reporter. ')
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/common/api.ts:151]:
```suggestion
    console.warn('⚠️ addEnvironment is deprecated and has no longer any functionality. Use reportedEnvironmentVars in wdio config instead. Read more in https://webdriver.io/docs/allure-reporter.')
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍 

@BorisOsipov any comments?

**Review by BorisOsipov — APPROVED** [review]

**Comment by BorisOsipov** [comment]
LGTM

**Comment by christian-bromann** [comment]
@m4hdyar mind signing the CLA?

**Comment by christian-bromann** [comment]
That should work, thanks!

**Pull request merged**